src/hooks/useMountState/useMountState.js

```diff
@@ -11,7 +11,7 @@ export const useMountState = () => {
     }
   }, [isMountedRef])
 
-  const isMounted = () => isMountedRef.current
+  const isMounted = useCallback(() => isMountedRef.current, [isMountedRef])
```

Comment:
using a ref in this code is kind of strange to me. if you like the ref use this isn't a big deal but i would have written this hook like:

```js
const [isMounted, setIsMounted] = useState(false)

useEffect(() => setIsMounted(true), [])

return { isMounted }
```

in general though having isMounted checks is a code smell for me so it may be more
worthwhile to try to refactor all the places this hook is needed than to worry about rewriting it right now.

Reply:
I'm glad you raised this because it made me realize that #526 introduced the inclusion of https://usehooks-ts.com/. This library provides [an implementation of `useIsMounted`](https://usehooks-ts.com/react-hook/use-is-mounted) which effectively abstracts away the contents of this hook. Ironically [the library implementation](https://github.com/juliencrn/usehooks-ts/blob/20667273744a22dd2cd2c48c38cd3c10f254ae47/packages/usehooks-ts/src/useIsMounted/useIsMounted.ts#L14-L26) seems equivalent to what we have here, but since it's in a library we don't have to worry too much about the implementation details so long as it works (which it does).

I've switched to using the `usehooks-ts` implementation in https://github.com/jeremyckahn/farmhand/commit/72d110b2155f1df3d7f631c27573997aa591da82 to eliminate the maintenance burden of our custom one.

> in general though having isMounted checks is a code smell for me so it may be more
worthwhile to try to refactor all the places this hook is needed than to worry about rewriting it right now.

I generally agree and try to avoid using `isMounted` checks for this reason. In this case it was necessary for Cow image generation: https://github.com/jeremyckahn/farmhand/blob/72d110b2155f1df3d7f631c27573997aa591da82/src/components/CowCard/CowCard.js#L129-L136

In this part of the code, the Cow image is generated dynamically as an asynchronous operation. By the time the image generation is complete, it's possible that the component has unmounted which would mean `setCowImage(cowImage)` would be improperly setting state on an unmounted component. There are probably cleaner abstractions for this sort of thing, but this seemed like the simplest solution. I don't know that we'd want to repeat this pattern too much, but I think this is the only part of the code base that we do this right now so I think that this bit of isolated awkwardness is acceptable.

Reply:
oh that's great we have a library version of this to use instead! also if it's only used for that one use case that's not a big deal at all. i did not look around for the uses